packages/dbml-parse/src/lib/interpreter/elementInterpreter/table.ts

```diff
@@ -141,14 +141,16 @@ export class TableInterpreter implements ElementInterpreter {
 
     column.token = getTokenPosition(field);
     column.inline_refs = [];
-    
+
     const settings = field.args.slice(1);
     if (_.last(settings) instanceof ListExpressionNode) {
       const settingMap = aggregateSettingList(settings.pop() as ListExpressionNode).getValue();
       column.pk = !!settingMap['pk']?.length || !!settingMap['primary key']?.length;
       column.increment = !!settingMap['increment']?.length;
       column.unique = !!settingMap['unique']?.length;
-      column.not_null = !!settingMap['not null']?.length && true;
+
+      const hasNullSetting = !!settingMap['null']?.length || !!settingMap['not null']?.length;
+      column.not_null = hasNullSetting ? !!settingMap['not null']?.length : undefined;
```

Comment:
I think this is clearer
```suggestion
      column.not_null = !!settingMap['not null']?.length ? true : !!settingMap['null']?.length ? false : undefined;
```

Reply:
Ok, I changed to nested ternary in 45b8a91